src/appimagetool.c

```diff
@@ -903,6 +908,33 @@ main (int argc, char *argv[])
                         printf("Will not guess update information since zsyncmake is missing\n");
                     }
                 }
+            } else if (github_repository != NULL && github_ref != NULL) {
+                printf("Running on GitHub Actions\n");
+                gchar *zsyncmake_path = g_find_program_in_path ("zsyncmake");
+                if (zsyncmake_path) {
+                    if (strstr(github_ref, "/pull/")) {
+                        printf("Will not calculate update information for GitHub because this is a pull request\n");
+                    } else {
+                        printf("Guessing update information based on $GITHUB_REPOSITORY=%s and $GITHUB_REF=%s\n", github_repository, github_ref);
+                        char buf[1024];
+                        gchar **parts = g_strsplit (github_repository, "/", 2);
+                        char* channel;
+                        if (github_ref != "" && strstr(github_ref, "/continuous/")) {
+                            channel = "latest";
```

Comment:
Doesn't that trigger some warnings? These statements would likely be allocated as static memory, and you're using a writable pointer. The latter point alone is potentially dangerous...

Reply:
@TheAssassin  Did I fix it now?